sdk/servicebus/service-bus/src/constructorHelpers.ts

```diff
@@ -23,8 +24,12 @@ import { SharedKeyCredential } from "./servicebusSharedKeyCredential";
  *       this is used to compute the exponentially increasing delays between retries. Default: 30000 milliseconds.
  *    - `timeoutInMs`: Amount of time in milliseconds to wait before the operation times out. This will trigger a retry if there are any
  *       retry attempts remaining. Minimum value: 60000 milliseconds.
+ * - `requestOptions`   : Options used when creating and sending HTTP requests for this operation. (Not applicable for the pure AMQP operations.)
+ *    - `timeout`   : The number of milliseconds a request can take before automatically being terminated. If provided, this timeout will be passed/used for the auth operations such as obtaining the token while using `@azure/identity` and Azure AD credentials.
  */
-export interface ServiceBusClientOptions {
+export interface ServiceBusClientOptions extends Pick<OperationOptionsBase, "requestOptions"> {
+  // TODO: Instead of picking "requestOptions" from OperationOptionsBase,
+  //       use the whole OperationOptionsBase and properly propagate the abortSignal and tracingOptions.
```

Comment:
It would be useful in this comment to explain why you can't do this now (ie, is there a constraint?)

Reply:
There is no constraint. Since there is quite an amount of work involved in plugging the tracing options into the underlying methods, I figured it could be tackled better in a future PR.

Reply:
### Opening up the discussion
Is it okay to pass the abortSignal, requestOptions and tracingOptions at the top-level client and propagate them?
We generally put them in the options only for the async operations.

If not provided at the root client level, users would have to pass the requestOptions for each and every method they use if needed to(Example: #13436).

@richardpark-msft @ramya-rao-a @chradek @bterlson @xirzec

Reply:
I think it's weird to have the constructor take `OperationOptions`, since from the name they are intended to be per operation. I'm not sure what an abortSignal passed to an entire client would be useful for (ending all outgoing operations at once?) Same with tracingOptions, since presumably you have different parent contexts per operation.

I also feel like the type math in the above is getting a bit wild, since you're using Pick on a type that is itself composed with Pick. Can we just declare the interfaces we support in a cleaner way without deriving it?